docs/src/further_topics/ugrid/data_model.rst

```diff
@@ -0,0 +1,361 @@
+Any higher dimensional element - an edge/face/volume - is described by a
+sequence of the indices of the nodes that make up that element. E.g. a
+triangular face made from connecting the first, third and fourth nodes:
+``[0, 2, 3]``. These 1D sequences combine into a 2D array enumerating **all**
+the elements of that type - edge/face/volume - called a **connectivity**.
+E.g. we could make a mesh of 4 nodes, with 2 triangles described using this
+``face_node_connectivity``: ``[[0, 2, 3], [3, 2, 1]]`` (note the shared nodes).
+
+.. note:: More on Connectivities:
+
+        * The element type described by a connectivity is known as its
+          **location**; ``edge`` in ``edge_node_connectivity``.
+        * According to the UGRID conventions, the nodes in a face should be
+          listed in "anti-clockwise order from above".
+        * Connectivities also exist to connect the higher dimensional elements,
+          e.g. ``face_edge_connectivity``. These are optional conveniences to
+          speed up certain operations and will not be discussed here.
+
+.. important::
+
+        **Meshes are unstructured**. Elements are enumerated along a single
+        **unstructured dimension** represented by either the coordinate or
+        connectivity arrays detailed above, and an element's position within
+        its respective array has nothing to do with its spatial position.
+
+A data variable associated with a mesh has a **location** of either ``node``,
+``edge``, ``face`` or ``volume``. The data is stored in a 1D array with one
+datum per element, matched to its element by matching the datum index with the
+coordinate or connectivity index. So for an example data array called ``foo``:
+``foo[3]`` would be at position ``(x[3], y[3])`` if it were node-located, or at
+``faces[3]`` if it were face-located.
+
+.. figure:: images/data_ugrid_mesh.svg
+   :alt: Diagram of how data is represented on an unstructured mesh
+
+   Data on an Unstructured Mesh
+
+   :download:`full size image <images/data_ugrid_mesh.svg>`
+
+----
+
+The mesh model also supports edges/faces/volumes having associated 'centre'
+coordinates - to allow point data to be assigned to these elements. 'Centre' is
+just a convenience term - the points can exist anywhere within their respective
+elements.
+
+.. figure:: images/ugrid_element_centres.svg
+   :alt: Diagram demonstrating mesh face-centred data.
+
+   Data can be assigned to mesh edge/face/volume 'centres'
+
+Mesh Flexibility
+++++++++++++++++
+Above we have seen how one could replicate data on a structured grid using
+a mesh instead. But the utility of a mesh is the extra flexibility it offers.
+Here are the main examples:
+
+* Every node is completely independent - every one can have unique X and
+  Y (and Z) coordinate values.
+
+.. figure:: images/ugrid_node_independence.svg
+   :alt: Diagram demonstrating the independence of each mesh node
+   :align: center
+   :width: 300
+
+   Every mesh node is completely independent
```

Comment:
I don't understand what this figure is intending to represent. Particularly the arrows. Are they to depicting the transition from old world (structured grids) to new world (unstructured grids)
![image](https://user-images.githubusercontent.com/8101163/151149891-149f9a76-9879-4bde-82f7-dc37d573502b.png)

Reply:
I wanted to introduce the new concepts gradually. So the initial introduction of the 'new world' was replicating the 'old world' example. I wanted to demonstrate that arrays of the **_same shape and size_** could also represent something unstructured.

I hope you'll agree that's an important step. But presumably there's a better way to represent this (or maybe just including the explanation I've just given you)?

Reply:
https://github.com/SciTools/iris/pull/4504#discussion_r793486314

Reply:
https://github.com/SciTools/iris/pull/4504/commits/3692c1cfa51f15181437b22088f3f223d0037a93